crates/wdk-build/src/lib.rs

```diff
@@ -472,118 +488,203 @@ impl Config {
+    /// dispatching based off of the [`Config`]. Returns `None` if the driver
+    /// model is [`DriverConfig::Wdm`]
+    #[must_use]
+    pub fn compute_wdffunctions_symbol_name(&self) -> Option<String> {
+        let (wdf_major_version, wdf_minor_version) = match self.driver_config {
+            DriverConfig::Kmdf(config) => {
+                (config.kmdf_version_major, config.target_kmdf_version_minor)
             }
-        }
+            DriverConfig::Umdf(config) => {
+                (config.umdf_version_major, config.target_umdf_version_minor)
+            }
+            DriverConfig::Wdm => return None,
+        };
 
-        Ok(())
+        Some(format!(
+            "WdfFunctions_{wdf_major_version:02}0{wdf_minor_version:02}"
+        ))
     }
 
-    /// Configures a Cargo build of a binary that depends on the WDK. This
+    /// Configure a Cargo build of a binary that depends on the WDK. This
     /// emits specially formatted prints to Cargo based on this [`Config`].
     ///
     /// This consists mainly of linker setting configuration. This must be
     /// called from a Cargo build script of the binary being built
     ///
     /// # Errors
     ///
-    /// This function will return an error if any of the required paths do not
-    /// exist.
+    /// This function will return an error if:
+    /// * any of the required WDK paths do not exist
+    /// * the C runtime is not configured to be statically linked
     ///
     /// # Panics
     ///
-    /// Panics if the invoked from outside a Cargo build environmen
+    /// Panics if the invoked from outside a Cargo build environment
     pub fn configure_binary_build(&self) -> Result<(), ConfigError> {
-        self.configure_library_build()?;
-
-        // Linker arguments derived from Microsoft.Link.Common.props in Ni(22H2) WDK
-        println!("cargo::rustc-cdylib-link-arg=/NXCOMPAT");
-        println!("cargo::rustc-cdylib-link-arg=/DYNAMICBASE");
-
-        // Always generate Map file with Exports
-        println!("cargo::rustc-cdylib-link-arg=/MAP");
-        println!("cargo::rustc-cdylib-link-arg=/MAPINFO:EXPORTS");
-
-        // Force Linker Optimizations
-        println!("cargo::rustc-cdylib-link-arg=/OPT:REF,ICF");
+        if !Self::is_crt_static_linked() {
+            return Err(ConfigError::StaticCrtNotEnabled);
+        }
 
-        // Enable "Forced Integrity Checking" to prevent non-signed binaries from
-        // loading
-        println!("cargo::rustc-cdylib-link-arg=/INTEGRITYCHECK");
+        let library_paths: Vec<PathBuf> = self.get_library_paths()?;
 
-        // Disable Manifest File Generation
-        println!("cargo::rustc-cdylib-link-arg=/MANIFEST:NO");
+        // Emit linker search paths
+        for path in library_paths {
+            println!("cargo::rustc-link-search={}", path.display());
+        }
 
         match &self.driver_config {
-            DriverConfig::WDM() => {
+            DriverConfig::Wdm => {
```

Comment:
Yes. Ideally we'd be able to consume the WDK's various xml files (vcxproj, targets, etc) and get all of this information. There'd still need to be code here to do things to emit the correctly formatted print statements to pass the information to cargo, but all that information would be parsed from the xmls instead of hardcoded here.

> Maybe I need to build a deserializer for the MSBuild proj files directly
There would need to be work done to be able to merge all the xmls together, since many of the msbuild files reference other xml files and merge them together in some deterministic fashion. I'd imagine merging all that together, and then deserializing that merged xml using some serde xml deserializer(ex. serde_xml_rs) could work.